src/parquet/encodings/dictionary-encoding.h

```diff
@@ -418,6 +432,23 @@ inline void DictEncoder<FixedLenByteArray>::WriteDict(uint8_t* buffer) {
   }
 }
 
+template<typename T>
+inline void DictEncoder<T>::FreeMemory() {}
+
+template<>
+inline void DictEncoder<ByteArray>::FreeMemory() {
+  for (const auto& value : uniques_) {
+    pool_->Free(const_cast<uint8_t*>(value.ptr));
+  }
+}
+
+template<>
+inline void DictEncoder<FixedLenByteArray>::FreeMemory() {
+  for (const auto& value : uniques_) {
+    pool_->Free(const_cast<uint8_t*>(value.ptr));
+  }
+}
```

Comment:
`ChunkedMemoryPool`is only one possible implementation of a memory pool. It does not free specified memory and cannot reuse freed memory. It is not a cure-all solution.
IMHO, we should only provide a memory pool interface (which includes at least `Allocate` and `Free` methods) and leave the implementation trade-offs to users. They can use the implemented `TrackingMemoryPool` or `ChunkedMemoryPool`, or implement their own pool.
To alleviate your concern, I propose to modify `get_default_pool()` to return `ChunkedMemoryPool` rather than `TrackedMemoryPool`.